src/nvim/getchar.c

```diff
@@ -1108,6 +1118,12 @@ static void gotchars(char_u *chars, size_t len)
       add_buff(&recordbuff, buf, 1L);
     }
   }
+
+  // remember how many chars were last recorded
+  if (Recording) {
+    last_recorded_len += num_bytes;
+  }
```

Comment:
I see one new test for checking that the change works with macros, but are there any tests you can think of to make sure this didn't cause a regression?